src/Kunstmaan/NodeSearchBundle/Services/SearchService.php

```diff
@@ -120,13 +125,29 @@ public function search()
         $request = $this->requestStack->getCurrentRequest();
 
         // Retrieve the current page number from the URL, if not present of lower than 1, set it to 1
+        $pageNumber = $this->getRequestedPage($request);
         $entity = $request->attributes->get('_entity');
 
-        $pageNumber = $this->getRequestedPage($request);
-        $searcher   = $this->container->get($entity->getSearcher());
+        if (!empty($this->searchers)) {
+            foreach ($this->searchers as $possibleSearcher) {
+                if ($entity->getSearcher() === \get_class($possibleSearcher)) {
```

Comment:
We may need to rethink this check logic. This won't work for users still using "old style" service id's in their custom code. I think it's better to still make a compiler pass and inject an array, with keys = serviceIds and value = the reference object, into this class. This will make it also easier to check if we have a tagged service for that searcher or not (if the serviceId as key exist, we are good to go).